neqo-udp/src/lib.rs

```diff
@@ -8,37 +8,133 @@
 #![allow(clippy::missing_panics_doc)] // Functions simply delegate to tokio and quinn-udp.
 
 use std::{
+    cell::RefCell,
     io::{self, IoSliceMut},
-    net::{SocketAddr, ToSocketAddrs},
+    net::SocketAddr,
     slice,
 };
 
-use neqo_common::{Datagram, IpTos};
+use neqo_common::{qdebug, Datagram, IpTos};
 use quinn_udp::{EcnCodepoint, RecvMeta, Transmit, UdpSocketState};
-use tokio::io::Interest;
 
 /// Socket receive buffer size.
 ///
 /// Allows reading multiple datagrams in a single [`Socket::recv`] call.
 const RECV_BUF_SIZE: usize = u16::MAX as usize;
 
-pub struct Socket {
+std::thread_local! {
+    static RECEIVE_BUFFER: RefCell<Vec<u8>> = RefCell::new(vec![0; RECV_BUF_SIZE]);
+}
+
+pub struct Socket<S> {
+    state: UdpSocketState,
     #[allow(unknown_lints)] // available with Rust v1.75
     #[allow(clippy::struct_field_names)]
-    socket: tokio::net::UdpSocket,
-    state: UdpSocketState,
-    recv_buf: Vec<u8>,
+    socket: S,
```

Comment:
Rather than suppress lints, maybe this can now just be:

```suggestion
    inner: S,
```

Reply:
Good idea. See https://github.com/mozilla/neqo/pull/1920/commits/8040a63b070df83191f03fd0697083ea438a1e21.